packages/vaex-contrib/vaex/contrib/io/gbq.py

```diff
@@ -0,0 +1,197 @@
+
+
+def to_table(df, dataset, table, job_config=None, client_project=None, credentials=None, chunk_size=None, progress=None):
+    '''Upload a Vaex DataFrame to a Google BigQuery Table.
+
+    Note that the upload creates a temporary parquet file on the local disk, which is then upload to
+    Google BigQuery.
+
+    :param DataFrame df: The Vaex DataFrame to be uploaded.
+    :param str dataset: The name of the dataset to which the table belongs
+    :param str table: The name of the table
+    :param job_config: Optional, an instance of google.cloud.bigquery.job.load.LoadJobConfig
+    :param str client_project: The ID of the project that executes the query. Will be passed when creating a job. If `None`, falls back to the default inferred from the environment.
+    :param credentials: The authorization credentials to attach to requests. See google.auth.credentials.Credentials for more details.
+    :param chunk_size: In case the local disk space is limited, export the dataset in chunks.
+                       This is considerably slower than a single file upload and it should be avoided.
+    :param progress: Valid only if chunk_size is not None. A callable that takes one argument (a floating point value between 0 and 1) indicating the progress, calculations are cancelled when this callable returns False
+
+    Example:
+
+    >>> import os
+    >>> os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = '../path/to/project_access_key.json'
+    >>> import vaex
+    >>> from vaex.contrib.io.gbq import to_table
+
+    >>> df = vaex.example()
+    >>> dataset = 'my_dataset'
+    >>> table = 'my_table'
+
+    >>> to_table(df=df, dataset=dataset, table=table)
+
+    '''
+
+    # Instantiate the BigQuery Client
+    client = google.cloud.bigquery.Client(project=client_project, credentials=credentials)
+
+    # Confirm configuration of the LoadJobConfig
+    if job_config is not None:
+        assert isinstance(job_config, google.cloud.bigquery.job.load.LoadJobConfig)
+        job_config.source_format = google.cloud.bigquery.SourceFormat.PARQUET
+    else:
+        job_config = google.cloud.bigquery.LoadJobConfig(source_format=google.cloud.bigquery.SourceFormat.PARQUET)
+
+    # Table to which to upload
+    table_bq = f"{dataset}.{table}"
+
+    if chunk_size is None:
+        with tempfile.NamedTemporaryFile(suffix='.parquet') as tmp:
+            df.export_parquet(tmp.name)
+            with open(tmp.name, "rb") as source_file:
+                job = client.load_table_from_file(source_file, table_bq, job_config=job_config)
+            job.result()
+
+    else:
+        progressbar = vaex.utils.progressbars(progress)
+        n_samples = len(df)
+
+        for i1, i2, table in df.to_arrow_table(chunk_size=chunk_size):
+            progressbar(i1 / n_samples)
+            with tempfile.NamedTemporaryFile(suffix='.parquet') as tmp:
+                pq.write_table(table, tmp.name)
+                with open(tmp.name, "rb") as source_file:
```

Comment:
This is not supported on windows https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile